gulp/docs.js

```diff
@@ -74,74 +69,185 @@ module.exports = {
-						// TODO: determine composition from the `import` statements See
-						// existing handlers for examples:
-						// https://github.com/reactjs/react-docgen/blob/dca8ec9d57b4833f7ddb3164bedf4d74578eee1e/src/handlers/propTypeCompositionHandler.js
-						var childComponentDocs = _.map(_.reject(_.keys(definitionMap), _.partial(_.isEqual, exportIdentiferName)), function (childComponentId) {
-							return getDocsForPath(definitionMap[childComponentId], childComponentId);
+					[function handler(documentation, definition) {
+
+						// component docs
+						if (findParentNodeIdentifier(definition) === exportName && isTopLevelDef(definition)) {
+
+							_.forEach(reactDocgen.defaultHandlers, function(handler) {
+								handler(documentation, definition);
+							});
+
+							return recast.visit(definition, {
+								visitObjectExpression: function (path) {
+									var isPrivateComponent = _.chain(path.get('properties').value)
+										.find(_.matchesProperty('key.name', '_lucidIsPrivate'))
+										.get('value.value', false)
+										.value();
+									documentation.set('isPrivateComponent', isPrivateComponent);
+									return false;
+								},
+							});
+
+						}
+
+						// childComponent docs
+
+						// import reference component
+						if (definition.value.type === 'Identifier') {
+							documentation.set('displayName', definition.parentPath.value.key.name);
+							documentation.set('description', '');
+							return documentation.set('componentRef', definition.value.name);
+						}
+
+						// import reference component's child component
+						if (definition.value.type === 'MemberExpression') {
+							documentation.set('displayName', definition.parentPath.value.key.name);
+							documentation.set('description', '');
+							return documentation.set('componentRef', `${definition.value.object.name}.${definition.value.property.name}`);
+						}
+
+						// list of default handlers
+						_.forEach([
+							reactDocgen.handlers.propTypeHandler,
+							reactDocgen.handlers.propTypeCompositionHandler,
+							reactDocgen.handlers.propDocBlockHandler,
+							reactDocgen.handlers.flowTypeHandler,
+							reactDocgen.handlers.flowTypeDocBlockHandler,
+							reactDocgen.handlers.defaultPropsHandler,
+							// use normal component docblock handler for top-level defined child components,
+							// otherwise this custom one
+							isTopLevelDef(definition)
+							? reactDocgen.handlers.componentDocblockHandler
+							: function(documentation, definition) {
+								var description = reactDocgen.utils.docblock.getDocblock(definition.parentPath.parentPath.parentPath) || '';
+								documentation.set('description', description);
+							},
+							reactDocgen.handlers.displayNameHandler,
+							reactDocgen.handlers.componentMethodsHandler,
+							reactDocgen.handlers.componentMethodsJsDocHandler,
+						], function(handler) {
+							handler(documentation, definition);
+						});
+
+						documentation.set(
```

Comment:
so this is a little weird because `displayName` here just means what we use to display in the docs, which i don't think is the same thing as the component `displayName` to be shown in react error stacks, etc. I was a little confused at first, too, and here I'm just sticking to what was already being done. I'm not sure what the right answer is.